light/store/db/db.go

```diff
@@ -220,15 +220,14 @@ func (s *dbs) LightBlockBefore(height int64) (*types.LightBlock, error) {
 //
 // Safe for concurrent use by multiple goroutines.
 func (s *dbs) Prune(size uint16) error {
-	// 1) Check how many we need to prune.
-	s.mtx.RLock()
-	sSize := s.size
-	s.mtx.RUnlock()
+	// 1) Check how many blocks we need to prune.
+	s.mtx.Lock()
```

Comment:
can we maybe add a comment explaining why we need to hold a lock for the whole duration?

Reply:
+ link an issue

Reply:
First off do you think it's possible / worth trying to do two different locks. 

I guess the risk is that the light client might perform a simultaneous deletion or addition hence I thought it might be better for the light store to lock on any write operation. But what do you think?

Reply:
I'm going to revert to having a separate read and write lock. I think this still makes the operation atomic

Reply:
> I guess the risk is that the light client might perform a simultaneous deletion or addition hence I thought it might be better for the light store to lock on any write operation. But what do you think?

I agree. It's safer that way 🦺  Sorry, I didn't see the second lock at the end the first time.